Potential overflow in multiply

I just ended up getting a panic on this line;

https://github.com/xiph/rav1e/blob/cbdf0703160bd4153975cb4ac3192476713f50c8/src/activity.rs#L99

Callstack;

```
[2022-04-02][02:10:07][panic][ERROR] thread 'main' panicked at 'attempt to multiply with overflow': C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\activity.rs:99
   0: backtrace::backtrace::dbghelp::trace
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.64\src\backtrace\dbghelp.rs:98
   1: backtrace::backtrace::trace_unsynchronized<backtrace::capture::impl$1::create::closure$0>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.64\src\backtrace\mod.rs:66
   2: backtrace::backtrace::trace<backtrace::capture::impl$1::create::closure$0>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.64\src\backtrace\mod.rs:53
   3: backtrace::capture::Backtrace::create
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.64\src\capture.rs:176
   4: backtrace::capture::Backtrace::new
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\backtrace-0.3.64\src\capture.rs:140
   5: log_panics::init::closure$0
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\log-panics-2.0.0\src\lib.rs:52
   6: std::panicking::rust_panic_with_hook
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\std\src\panicking.rs:610
   7: std::panicking::begin_panic_handler::closure$0
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\std\src\panicking.rs:500
   8: std::sys_common::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure$0,never$>
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\std\src\sys_common\backtrace.rs:139
   9: std::panicking::begin_panic_handler
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\std\src\panicking.rs:498
  10: core::panicking::panic_fmt
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\core\src\panicking.rs:116
  11: core::panicking::panic
             at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a\/library\core\src\panicking.rs:48
  12: rav1e::activity::variance_8x8<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\activity.rs:99
  13: rav1e::activity::ActivityMask::from_plane<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\activity.rs:51
  14: rav1e::api::internal::ContextInner<u16>::encode_packet<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\api\internal.rs:1181
  15: rav1e::api::internal::ContextInner<u16>::receive_packet<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\api\internal.rs:1330
  16: rav1e::api::context::impl$0::receive_packet::closure$0<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\api\context.rs:295
  17: rav1e::api::context::Context<u16>::receive_packet<u16>
             at C:\Users\Jasper\.cargo\registry\src\github.com-1ecc6299db9ec823\rav1e-0.5.1\src\api\context.rs:299
  18: video_encode_test::main
```

Not quite sure that the exact input is - but presumably this needs to do some kind of saturating / overflowing mul instead?

(Running with debug-features=false to skip the nasm build time requirement fwiw)

--- a/src/activity.rs
+++ b/src/activity.rs
@@ -92,11 +92,11 @@ fn variance_8x8<T: Pixel>(src: &PlaneRegion<'_, T>) -> u32 {
   }
 
   // Sum together the sum of columns
-  let sum_s = sum_s_cols.iter().map(|&a| u32::cast_from(a)).sum::<u32>();
-  let sum_s2 = sum_s2_cols.iter().sum::<u32>();
+  let sum_s = sum_s_cols.iter().copied().map(u64::from).sum::<u64>();
+  let sum_s2 = sum_s2_cols.iter().copied().map(u64::from).sum::<u64>();
 
   // Use sums to calculate variance
-  sum_s2 - ((sum_s * sum_s + 32) >> 6)
+  u32::try_from(sum_s2 - ((sum_s * sum_s + 32) >> 6)).unwrap_or(u32::MAX)
 }
 
 /// `rsqrt` result stored in fixed point w/ scaling such that:

